client/src/views/client-area/DashboardPage.vue

```diff
@@ -1,18 +1,452 @@
+      <div class="dashboard-section-title">
+        <ion-title class="title-h2">{{ $msTranslate('clientArea.dashboard.payment.title') }}</ion-title>
+        <ion-text class="payment-text body">
+          {{ $msTranslate('clientArea.dashboard.payment.description') }}
+        </ion-text>
+      </div>
+      <div class="payment">
+        <ms-stripe-card-details
+          v-if="defaultCard"
+          :card="defaultCard"
+          class="payment-card"
+        />
+        <ion-text
+          class="custom-button custom-button-fill button-medium"
+          @click="switchPage(ClientAreaPages.PaymentMethods)"
+        >
+          {{ $msTranslate('clientArea.dashboard.payment.update') }}
+        </ion-text>
+      </div>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-import { BmsOrganization } from '@/services/bms';
+import { IonTitle, IonText, IonList, IonItem, IonIcon } from '@ionic/vue';
+import { download } from 'ionicons/icons';
+import { ClientAreaPages } from '@/views/client-area/types';
+import {
+  BmsAccessInstance,
+  DataType,
+  BmsOrganization,
+  OrganizationStatsResultData,
+  BmsInvoice,
+  BillingDetailsResultData,
+  BillingDetailsPaymentMethodCard,
+  PaymentMethod,
+} from '@/services/bms';
+import { MsInformationTooltip, I18n, MsStripeCardDetails } from 'megashark-lib';
+import { onMounted, ref } from 'vue';
+import { PaymentMethod as MsPaymentMethod } from 'megashark-lib';
 
-defineProps<{
+const props = defineProps<{
   organization: BmsOrganization;
 }>();
+
+const stats = ref<OrganizationStatsResultData | undefined>(undefined);
+const invoices = ref<Array<BmsInvoice>>([]);
+const billingDetails = ref<BillingDetailsResultData | undefined>(undefined);
+const defaultCard = ref<MsPaymentMethod.Card | undefined>(undefined);
+const currentPage = ref<ClientAreaPages>(ClientAreaPages.Dashboard);
+const currentMonthYear = new Date().toLocaleDateString(I18n.getLocale(), { month: 'long', year: 'numeric' });
+const currentDate = new Date();
+
+// should be fetched from the backend
+const ADMIN_USER_PRICE = 15;
+
+onMounted(async () => {
+  const orgStatsResponse = await BmsAccessInstance.get().getOrganizationStats(props.organization.bmsId);
+  if (!orgStatsResponse.isError && orgStatsResponse.data && orgStatsResponse.data.type === DataType.OrganizationStats) {
+    stats.value = orgStatsResponse.data;
+  }
+
+  const invoicesResponse = await BmsAccessInstance.get().getInvoices();
+  if (!invoicesResponse.isError && invoicesResponse.data && invoicesResponse.data.type === DataType.Invoices) {
+    invoices.value = invoicesResponse.data.invoices.filter((invoice) => invoice.organizationId === props.organization.parsecId);
+  }
+
+  const billingDetailsResponse = await BmsAccessInstance.get().getBillingDetails();
+  if (!billingDetailsResponse.isError && billingDetailsResponse.data && billingDetailsResponse.data.type === DataType.BillingDetails) {
+    billingDetails.value = billingDetailsResponse.data;
```

Comment:
Are you using the `billingDetails` variable? Might as well remove it, you're only interested in the active card.